sourcery_analytics/metrics/utils.py

```diff
@@ -38,12 +38,30 @@ def method_name(method: astroid.nodes.FunctionDef) -> str:
 @nodedispatch
 @validate_node_type(astroid.nodes.FunctionDef)
 def method_lineno(method: astroid.nodes.FunctionDef) -> int:
+    """Returns the line number of the method.
+
+    Not very useful by itself, but can be combined with other metrics for convenience.
+
+    Examples:
+        >>> method_lineno("def foo(): pass")
+        1
+    """
     return method.lineno
 
 
 @nodedispatch
 @validate_node_type(astroid.nodes.FunctionDef)
 def method_file(method: astroid.nodes.FunctionDef) -> str:
+    """Returns the file name the method is in.
```

Comment:
**issue (complexity):** Consider simplifying the implementation and documentation of `method_lineno` and `method_file` functions.

The addition of `method_lineno` and `method_file` functions introduces a bit more complexity than necessary, particularly with the extensive documentation and the nested function definition within `method_file`. While the documentation is thorough, for straightforward functions like these, a simpler approach might be more in line with the existing code patterns. For instance, the nested `get_module` function in `method_file` could be avoided by directly iterating through the parent nodes until the module node is found. This would streamline the function and maintain consistency with the simpler, more direct style seen in other parts of the codebase. Simplifying the documentation to be more concise could also help in keeping the overall codebase easy to read and maintain.